### PR TITLE
add overview of rolling upgrade

### PR DESCRIPTION
**Pull request by DanRoscigno**
Rolling upgrade for Self-managed ClickHouse server

### Changelog category (leave one):
- Documentation (changelog entry is not required)

### CODE REVIEW

**Review by genzgd — APPROVED** [review]

**Review by tavplubix — APPROVED** [review]

**Comment by tavplubix** [comment]
Maybe we should also mention that rolling upgrade is not recommended when the difference between versions is more than one year or there is more than 1 LTS version between them. And that it's not recommended to keep different versions running together for a long time.

Ref: https://github.com/ClickHouse/clickhouse-docs/issues/548

**Comment by DanRoscigno** [comment]
> Maybe we should also mention that rolling upgrade is not recommended when the difference between versions is more than one year or there is more than 1 LTS version between them. And that it's not recommended to keep different versions running together for a long time.
> 
> Ref: [ClickHouse/clickhouse-docs#548](https://github.com/ClickHouse/clickhouse-docs/issues/548)

Thanks! I knew that there were notes about this somewhere, but I was looking in the wrong spot.  I will incorporate your notes and push.

File: docs/en/operations/update.md [hunk 1]
```diff
@@ -1,10 +1,75 @@
...
+
+
+### Test experimental features on a separate ClickHouse server running the target version
+
+The compatibility of experimental features can be broken at any moment in any way.  If you are using experimental features, then check the changelogs and consider setting up a separate ClickHouse server with the target version installed and test your use of the experimental features there.
+
+### Downgrades
+If you upgrade and then realize that the new version is not compatible with some feature that you depend on you may be able to downgrade to a recent (less than one year old) version if you have not started to use any of the new features.  Once the new features are used the downgrade will not work.
+
+### Minor version upgrades
```
Comment (tavplubix) [inline docs/en/operations/update.md:32]: Did you mean "Major version upgrades" here?

Comment (DanRoscigno) [reply docs/en/operations/update.md:32]: I am in the process of changing that title to:

### Multiple ClickHouse server versions in a cluster 

and adding in the warnings that you added to the issue.

Comment (DanRoscigno) [reply docs/en/operations/update.md:32]: updated in https://github.com/ClickHouse/ClickHouse/pull/44083

**Comment by Felixoid** [comment]
Style check is broken after this PR. You didn't wait for the results after the latest `update branch`. Now CI is broken for every new PR

**Comment by DanRoscigno** [comment]
> Style check is broken after this PR. You didn't wait for the results after the latest `update branch`. Now CI is broken for every new PR
> 
> https://s3.amazonaws.com/clickhouse-test-reports/44079/5b54bb1b3112a8c139cb1033a49c14ad797e1d4a/style_check/docs_spelling_output.txt

I do not understand, how did merging this PR break other ones?  Does the spell checker check all of the repo content when it is running against a single PR?

**Comment by alexey-milovidov** [comment]
@DanRoscigno @tavplubix 

The terminology is Major.Minor.Patch

**Pull request merged**